icepyx/core/auth.py

```diff
@@ -0,0 +1,155 @@
+        self._s3_initial_ts = None  # timer for 1h expiration on s3 credentials
+
+    def __str__(self):
+        if self.session:
+            repr_string = "EarthdataAuth obj with session initialized"
+        else:
+            repr_string = "EarthdataAuth obj without session initialized"
+        return repr_string
+
+    @property
+    def auth(self):
+        '''
+        Authentication object returned from earthaccess.login() which stores user authentication. 
+        '''
+        # Only login the first time .auth is accessed
+        if self._auth is None:
+            auth = earthaccess.login()
+            # check for a valid auth response
+            if auth.authenticated is False:
+                raise AuthenticationError('Earthdata authentication failed. Check output for error message')
+            else:
+                self._auth = auth
+                
+        return self._auth
+
+    @property
+    def session(self):
+        '''
+        Earthaccess session object for connecting to Earthdata resources.
+        '''
+        # Only generate a session the first time .session is accessed
+        if self._session is None:
+            self._session = self.auth.get_session()
+        return self._session
+
+    @property
+    def s3login_credentials(self):
+        '''
+        A dictionary which stores login credentials for AWS s3 access. This property is accessed
+        if using AWS cloud data.
+        '''
+        
+        def set_s3_creds():
+            ''' Store s3login creds from `auth`and reset the last updated timestamp'''
+            self._s3login_credentials = self.auth.get_s3_credentials(daac="NSIDC")
+            self._s3_initial_ts = datetime.datetime.now()
+            
+        # Only generate s3login_credentials the first time credentials are accessed, or if an hour
+        # has passed since the last login    
+        if self._s3login_credentials is None:
+            set_s3_creds()
+        elif (datetime.datetime.now() - self._s3_initial_ts) >= datetime.timedelta(hours=1):
+            set_s3_creds()
+        return self._s3login_credentials
+
+    def earthdata_login(self, uid=None, email=None, s3token=False, **kwargs) -> None:
+        """
+        Authenticate with NASA Earthdata to enable data ordering and download.
+
+        Generates the needed authentication sessions and tokens, including for cloud access.
+        Authentication is completed using the [earthaccess library](https://nsidc.github.io/earthaccess/).
+        Methods for authenticating are:
+            1. Storing credentials as environment variables ($EARTHDATA_LOGIN and $EARTHDATA_PASSWORD)
+            2. Entering credentials interactively
+            3. Storing credentials in a .netrc file (not recommended for security reasons)
+        More details on using these methods is available in the [earthaccess documentation](https://nsidc.github.io/earthaccess/tutorials/restricted-datasets/#auth).
```

Comment:
I'd like to move this (or something like it) into the mixin docstring - users will still need to know how they can authenticate. This way that guidance won't be lost when we (ideally) someday deprecate this function.